nodestream/model/graph_objects.py

```diff
@@ -18,22 +22,42 @@ def update(self, other: "DeduplicatableObject"):
         raise NotImplementedError
 
 
+# This is a weird little trick to cache the current time for a short period of
+# time. When nodestream is adding the timestamp to tones of node and
+# relationship objects, it can be a bit slow. Just to get times that are
+# basically the same, there is little value in getting the current time over
+# and over again. So we cache the current time for a short period of time.
+# Profiling this generally shows a 5-15% speedup in the time it takes to
+# ingest a large number of nodes and relationships (Excluding database time).
+
+
+@lru_cache(maxsize=128)
+def _get_cached_timestamp(time_hash: int) -> Timestamp:
+    del time_hash  # Unused; just needed to cache the result
+    return Timestamp.utcnow()
+
+
+def get_cached_timestamp(max_age_in_seconds: int = 2) -> Timestamp:
+    # calling time.time() is relatively fast, but getting a Timestamp object
+    # is relatively slow.
+    ttl_hash = round(time.time()) / max_age_in_seconds
```

Comment:
not sure I understand what max_age_in_seconds does...
If time = 3
ttl_hash = 1.5

If time = 4
ttl_hash = 2

From what I understand then you will get a different timestamp?

Reply:
Ah good catch. The reason this doesn't make sense is the round is supposed to be around the whole operation `time.time() / max_age_in_seconds`. In your scenario, 

If time = 3
ttl_hash = 1 

if time = 4 
ttl_hash=2

Reply:
btw round 1.5 = 2. But yes that makes more sense.